docs/source/design/github_actions.rst

```diff
@@ -0,0 +1,118 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_github_action:
+
+*******************************
+GitHub Action
+*******************************
+
+.. topic:: Specification scope and status
+
+   This specification describes a proposed interface to a DataLad GitHub Action.
+
+The purpose of the DataLad GitHub Action is to support CI testing with DataLad datasets
+by making it easy to install ``datalad`` and ``get`` data from the dataset.
+
+
+Example Usage
+=============
+
+Dataset installed at ``${GITHUB_WORKSPACE}/studyforrest-data-phase2``,
+``get``'s all the data::
+
+    - uses: datalad/datalad-action@master
+      with:
+        datasets:
+          - source: https://github.com/psychoinformatics-de/studyforrest-data-phase2
+          - install_get_data: true
+
+Specify advanced options::
+
+    - name: Download testing data
+      uses: datalad/datalad-action@master
+      with:
+        datalad_version: ^0.15.5
+        add_datalad_to_path: false
+        datasets:
+          - source: https://github.com/psychoinformatics-de/studyforrest-data-phase2
+          - install_path: test_data
```

Comment:
Let's add optional `branch` option to be able to checkout a specific branch or tag.  Might also want `--depth` but not sure how that would play with `git-annex` branch -- didn't experiment.

A little more of history/details: we at large didn't uniformly address https://github.com/datalad/datalad/issues/2109 in being able to install dataset at a given committish.

There was https://github.com/datalad/datalad/pull/5662 which was superseded with https://github.com/datalad/datalad/pull/6218 so it became possible to pass git options to `datalad clone` so it is possible to use `datalad clone URL --branch BRANCH` to point to specific branch or tag. We have also explicit `--branch` option in `datalad install`. In either of those cases that still does not allow to point to specific commit hexsha.  Since besides `--branch` , the `--depth` and may be other git options could come handy, may  be it would be better to code action as sequence of `datalad clone {url} [{path}] [{git_options}]` , `datalad -C {path} get` with `--no-data` and  `-r` options depending on settings instead of a single `datalad install` command.

Reply:
:+1: done.